### PR TITLE
ADR for deploying pipelines along with operator

### PR DESCRIPTION
**Pull request by harshad16**
ADR for deploying pipelines along with operator
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

## Related Issues and Dependencies
Fixes: https://github.com/thoth-station/opendatahub-cnbi/issues/10

### CODE REVIEW

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:12]:
```suggestion
CNBi feature include OpenShift Pipelines, that are used for task actions.
The pipelines are not entirely dependent on the CNBi Operator.
However the Operator trigger the pipelines execution, so there is a dependency from Operator end.
```
I think the mardown expect one phrase per line without line length limiting

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:27]:
```suggestion
Chosen option: "Include the pipelines with the Operator manifests", because it would provide consistency on deployment.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:23]:
```suggestion
* Include the pipelines with the Operator manifests.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:21]:
```suggestion
* Deploy the OpenShift pipelines using Helm charts, and maintain them on different repository.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:32]:
```suggestion
* Including the pipelines with the Operator simplify packaging and referencing. 
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:36]:
```suggestion
* This makes the pipelines dependent on the operator.
```
(Does it really though ? Can't we still just apply them as manifests ?

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:44]:
```suggestion
* Bad, because it requires versioning of helm chart and keeping track of what getting deployed with respect to  the Operator version. 
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:48]:
```suggestion
* Good, because it provide easy packaging along with operator.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:49]:
```suggestion
* Bad, because it would require Devops to continuously release the pipelines repository.
```
(That's a guess for devops

**Review by VannTen — CHANGES_REQUESTED** [review]

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md:35]: yes we can apply them directly, it is just that they would be in the operator repo.
so release of them is basically the release of operator.

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

File: docs/decisions/0002-deploy-pipelines.md
Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: thanks :+1:

**Comment by harshad16** [comment]
Thanks for the review
Have made all the changes.
Ready for review again.

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [inline docs/decisions/0002-deploy-pipelines.md:26]:
```suggestion
Chosen option: "Include the pipelines with the Operator manifests", because it provides consistency on deployment.
```

**Review by VannTen — CHANGES_REQUESTED** [review]
Just a little thing I missed this morning.

**Comment by harshad16** [comment]
Thanks for the reviews.
Updated the PR with more details.
So it could be deployed via kfdef.

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:9]:
```suggestion
The Custom Notebook Image (CNBi) feature of Open Data Hub (ODH) includes OpenShift Pipelines manifests that are used to implement the CNBi functionality.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:10]:
```suggestion
The pipelines are not entirely dependent on the [CNBi Operator](https://github.com/thoth-station/meteor-operator).
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:11]:
```suggestion
However the CNBi Operator triggers the pipelines execution, so there is a dependency from CNBi Operator's end.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:16]: It is a bit unclear (to me at least) what "minimize resource location" means...

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:20]:
```suggestion
* Deploy the OpenShift pipeline definition manifests using Helm charts, and maintain them on a different repository.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:21]:
```suggestion
* Have the Operator deploy the pipeline manifests using the tarball of a separate pipeline repository.
```

File: docs/decisions/0002-deploy-pipelines.md
Comment (codificat) [inline docs/decisions/0002-deploy-pipelines.md:22]:
```suggestion
* Include the pipeline manifests within the CNBi Operator repository, allow users to deploy it using kfdef.
```

**Review by codificat — COMMENTED** [review]
I have a few minor comments/suggestions, mostly things that sound "better" to me - but it might be just me, so please feel free to accept as appropriate.

Not approving yet so you can check these before merging, but feel free to self-approve when happy

File: docs/decisions/0002-deploy-pipelines.md
Comment (VannTen) [reply docs/decisions/0002-deploy-pipelines.md:16]:
```suggestion
* Avoid spreading resources in several locations
```
Maybe ?

Comment (harshad16) [reply docs/decisions/0002-deploy-pipelines.md]: yes, it means we don't want to maintain them separate in different places.

**Comment by codificat** [comment]
thanks!
/lgtm
/approve

**Pull request merged**
